model/oauth/client.go

```diff
@@ -426,6 +446,20 @@ func (c *Client) Update(i *instance.Instance, old *Client) *ClientRegistrationEr
 		c.NotificationDeviceToken = old.NotificationDeviceToken
 	}
 
+	// Updating metadata
+	appSlug := c.SoftwareID
+	if strings.HasPrefix(c.SoftwareID, "registry://") {
+		appSlug = strings.TrimPrefix(c.SoftwareID, "registry://")
+	}
+
+	if old.Metadata == nil {
+		md, _ := metadata.NewWithApp(appSlug, c.SoftwareVersion, DocTypeVersion)
```

Comment:
To be consistent, we should not set the `CreatedByApp` and `CreatedByAppVersion` fields if the SoftwareID is not linked to a registry webapp.